src/Interpreters/InterpreterSelectQueryAnalyzer.cpp

```diff
@@ -74,60 +75,28 @@ ContextMutablePtr buildContext(const ContextPtr & context, const SelectQueryOpti
-
-        auto & query_to_replace_table_expression_typed = query_to_replace_table_expression->as<QueryNode &>();
-        auto left_table_expression = extractLeftTableExpression(query_to_replace_table_expression_typed.getJoinTree());
-        auto left_table_expression_node_type = left_table_expression->getNodeType();
-
-        switch (left_table_expression_node_type)
-        {
-            case QueryTreeNodeType::QUERY:
-            case QueryTreeNodeType::UNION:
-            {
-                query_to_replace_table_expression = std::move(left_table_expression);
-                break;
-            }
-            case QueryTreeNodeType::TABLE:
-            case QueryTreeNodeType::TABLE_FUNCTION:
-            case QueryTreeNodeType::IDENTIFIER:
-            {
-                table_expression_to_replace = std::move(left_table_expression);
-                break;
-            }
-            default:
-            {
-                throw Exception(ErrorCodes::UNSUPPORTED_METHOD,
-                    "Expected table, table function or identifier node to replace with storage. Actual {}",
-                    left_table_expression->formatASTForErrorMessage());
-            }
-        }
-    }
+        auto & table_node = node->as<TableNode &>();
 
-    /// Don't replace storage if table name differs
-    if (auto * table_node = table_expression_to_replace->as<TableNode>(); table_node && table_node->getStorageID().getFullNameNotQuoted() != storage->getStorageID().getFullNameNotQuoted())
-        return;
+        /// Don't replace storage if table name differs
+        if (table_node.getStorageID().getFullNameNotQuoted() != storage->getStorageID().getFullNameNotQuoted())
+            continue;
 
-    auto replacement_table_expression = std::make_shared<TableNode>(storage, context);
-    std::optional<TableExpressionModifiers> table_expression_modifiers;
+        auto replacement_table_expression = std::make_shared<TableNode>(storage, context);
 
-    if (auto * table_node = table_expression_to_replace->as<TableNode>())
-        table_expression_modifiers = table_node->getTableExpressionModifiers();
-    else if (auto * table_function_node = table_expression_to_replace->as<TableFunctionNode>())
-        table_expression_modifiers = table_function_node->getTableExpressionModifiers();
-    else if (auto * identifier_node = table_expression_to_replace->as<IdentifierNode>())
-        table_expression_modifiers = identifier_node->getTableExpressionModifiers();
+        if (auto table_expression_modifiers = table_node.getTableExpressionModifiers())
+            replacement_table_expression->setTableExpressionModifiers(*table_expression_modifiers);
 
-    if (table_expression_modifiers)
-        replacement_table_expression->setTableExpressionModifiers(*table_expression_modifiers);
+        if (node->hasAlias())
+            replacement_table_expression->setAlias(node->getAlias() + "_replacement");
```

Comment:
Why do we need to add the suffix to alias?

Reply:
It'll be set to the alias of `node` anyway in `cloneAndReplace`. I don't think we need this line.